src/common/config/metadata/value.go

```diff
@@ -0,0 +1,141 @@
+	Value string `json:"value,omitempty"`
+}
+
+// NewConfigureValue ...
+func NewConfigureValue(key, value string) *ConfigureValue {
+	result := &ConfigureValue{}
+	if _, ok := Instance().GetMetaData(key); ok {
+		err := result.Set(key, value)
+		if err == nil {
+			return result
+		}
+		log.Errorf("Failed to set key:%v, value:%v, error %v", key, value, err)
+	} else {
+		log.Errorf("Failed to create a configure value without metadata, key %v, value %v", key, value)
+	}
+	return result
+}
+
+// GetString - Get the string value of current configure
+func (c *ConfigureValue) GetString() string {
+	// Any type has the string value
+	if _, ok := Instance().GetMetaData(c.Key); ok {
+		return c.Value
+	}
+	return ""
+}
+
+// GetKey ...
+func (c *ConfigureValue) GetKey() string {
+	return c.Key
+}
+
+// GetInt - return the int value of current value
+func (c *ConfigureValue) GetInt() int {
+	if item, ok := Instance().GetMetaData(c.Key); ok {
+		if intValue, suc := item.ItemType.get(c.Value).(int); suc {
+			return intValue
+		}
+	}
+	log.Errorf("The current value can not convert to integer, %+v", c)
+	return 0
+}
+
+// GetInt64 - return the int64 value of current value
+func (c *ConfigureValue) GetInt64() int64 {
+	if item, ok := Instance().GetMetaData(c.Key); ok {
+		if int64Value, suc := item.ItemType.get(c.Value).(int64); suc {
+			return int64Value
+		}
+	}
+	log.Errorf("Current value can not convert to int64, %+v", c)
+	return 0
+}
+
+// GetBool - return the bool value of current setting
+func (c *ConfigureValue) GetBool() bool {
+	if item, ok := Instance().GetMetaData(c.Key); ok {
+		if boolValue, suc := item.ItemType.get(c.Value).(bool); suc {
+			return boolValue
+		}
+	}
+	log.Errorf("The current value can not convert to bool, %+v", c)
+	return false
+}
+
+// GetStringToStringMap - return the string to string map of current value
+func (c *ConfigureValue) GetStringToStringMap() map[string]string {
+	result := map[string]string{}
+	if item, ok := Instance().GetMetaData(c.Key); ok {
+		if mapValue, suc := item.ItemType.get(c.Value).(map[string]string); suc {
+			return mapValue
+		}
+	}
+	log.Errorf("The current value can not convert to map[string]string, %+v", c)
+	return result
+}
+
+// Validate - to validate configure items, if passed, return true, else return false and return error
+func (c *ConfigureValue) Validate() error {
+	if item, ok := Instance().GetMetaData(c.Key); ok {
+		return item.ItemType.validate(c.Value)
+	}
+	return ErrNotDefined
+}
+
+// GetPassword ...
+func (c *ConfigureValue) GetPassword() string {
+	if _, ok := Instance().GetMetaData(c.Key); ok {
+		return c.Value
+	}
+	log.Errorf("Failed to get password, metadata not defined: %v", c.Key)
+	return ""
+}
+
+// Set - set this configure item to configure store
+func (c *ConfigureValue) Set(key, value string) error {
+	if item, ok := Instance().GetMetaData(key); ok {
+		if item.ItemType.validate(value) == nil {
+			c.Key = key
+			c.Value = value
+			return nil
+		}
```

Comment:
if `validate` returns a non-nil error, the `Set` will return ErrNotDefined.

This is incorrect.